src/nvim/lua/vim.lua

```diff
@@ -1,2 +1,62 @@
 -- TODO(ZyX-I): Create compatibility layer.
-return {}
+--{{{1 package.path updater function
+-- Last inserted paths. Used to clear out items from package.[c]path when they
+-- are no longer in &runtimepath.
+local last_nvim_paths = {}
+local function _update_package_paths()
+  local new_last_nvim_paths = {}
+  local rtps = vim.api.nvim_list_runtime_paths()
+  for _, key in ipairs({'path', 'cpath'}) do
+    local orig_str = package[key] .. ';'
+    local pathtrails = {}
+    local pathtrails_ordered = {}
+    local orig = {}
+    local orig_paths = {}
```

Comment:
Leftover from some of the previous implementations: specifically before I decided to have

```
        -- Additionally keep paths from &runtimepath at the start:
        -- unconditionally add them above, do not add them here.
        cur_nvim_paths[orig_path]
```

and keep previous order.

That was decided because if you reorder &runtimepath previous code would not reorder `package.[c]path`.